### PR TITLE
Release v2.0.0b2

### PR DESCRIPTION
**Pull request by jdrew82**
This PR is to prepare for cutting release for v2.0.0-beta2. 

- Updated version to specify beta2 to match Nautobot.
- Added release notes to summarize changes.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "nautobot-ssot"
-version = "2.0.0"
+version = "2.0.0-beta.2"
```
Comment (Kircheneer) [inline pyproject.toml:3]: Shouldn't this be beta 1 of SSoT?

File: docs/admin/release_notes/version_2.0.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+# v2.0 Release Notes
+
+## v2.0.0-beta.2 - 2023-07-13
```
Comment (Kircheneer) [inline docs/admin/release_notes/version_2.0.md:4]: Same here

**Review by Kircheneer — CHANGES_REQUESTED** [review]
Just the version number

File: pyproject.toml (hunk 1 above)
Comment (jdrew82) [reply pyproject.toml:3]: I was doing this to match Nautobot but I'll change it to beta 1. Will also add a compatibility matrix to documentation so it's clear what version is supported for which Nautobot version.

**Review by Kircheneer — APPROVED** [review]
LGTM!

**Pull request merged**